application/lib/view/screens/ScreenEditorScreen.dart

```diff
@@ -0,0 +1,245 @@
+import 'dart:async';
+
+import 'package:application/blocs/ScreenBloc.dart';
+import 'package:application/di.dart';
+import 'package:application/model/EditorScreenElement.dart';
+import 'package:application/model/Screen.dart';
+import 'package:application/routes.dart';
+import 'package:application/view/screens/BaseScreen.dart';
+import 'package:application/view/widgets/NotifyDIalog.dart';
+import 'package:application/view/widgets/RoundedTextField.dart';
+import 'package:flutter/material.dart';
+import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+
+class ScreenEditorScreen extends BaseScreen {
+  ScreenEditorScreen(this.screen) {
+    for (var widget in screen.screenContent) {
+      screenContent.add(EditorScreenElement(
+          widgetType: widget['type'],
+          key: widget['key'],
+          position: int.parse(widget['position'])));
+    }
```

Comment:
This will brake in the long run - how will you handle keys that one widget has that the other widgets dont have ?

TL;DR - Use factory pattern instead of this !

To fix this i would recommend that `EditorScreenElement` becomes an abstract class with a static build method and then you create a class per Widget type that inherits from `EditorScreenElement`. In this for-loop you can then send the entire json as `screenContent.add(EditorScreenElement.build(widget))` - The factory class will then check type to see which concrete class to new up, and on that you can have a `fromJson`-method that knows how to convert json into it own data properties..